rotkehlchen/tests/api/test_yearn_vaults.py

```diff
@@ -82,9 +95,51 @@ def test_query_yearn_vault_v2_balances(rotkehlchen_api_server, ethereum_accounts
         else:
             result = assert_proper_sync_response_with_result(response)
 
-    for vault in result[TEST_V2_ACC2].values():
-        assert '%' in vault['roi']
-        assert FVal(vault['vault_value']['amount']) > ZERO
-        assert FVal(vault['vault_value']['usd_value']) > ZERO
-        assert FVal(vault['underlying_value']['amount']) > ZERO
-        assert FVal(vault['underlying_value']['usd_value']) > ZERO
+    assert_proper_vault_response(result[TEST_V2_ACC2])
+
+
+@pytest.mark.vcr(filter_query_parameters=['apikey'])
+@pytest.mark.parametrize('ethereum_accounts', [[TEST_V3_ACC3]])
+@pytest.mark.parametrize('ethereum_modules', [['yearn_vaults_v3']])
+@pytest.mark.parametrize('should_mock_current_price_queries', [True])
+@pytest.mark.parametrize('should_mock_price_queries', [True])
+@pytest.mark.parametrize('default_mock_price_value', [ONE])
+def test_query_yearn_vault_v3_balances(
+        rotkehlchen_api_server: 'APIServer',
+        ethereum_accounts: list['ChecksumEvmAddress'],
+):
+    async_query = random.choice([True, False])
+    rotki = rotkehlchen_api_server.rest_api.rotkehlchen
+    add_yearn_v3_vault_token(
+        database=rotki.data.db,
+        address=string_to_evm_address('0x028eC7330ff87667b6dfb0D94b954c820195336c'),
+        name='DAI yVault',
+        symbol='yvDAI-1',
+        started=Timestamp(1710272855),
+        underlying_token=string_to_evm_address('0x6B175474E89094C44Da98b954EedeAC495271d0F'),
+    )
```

Comment:
can you define the function `add_yearn_v3_vault_token` in the decoder test file? Since you use it only there  unless I'm mistaken